P1: CRUD usage examples moved to appropriate pages (#618)
mongodb/docs-java: source/crud/read-operations/retrieve.txt

specify a query, see our :doc:`Specify a Query
</crud/query-document>` guide.

To retrieve a single document, you can append the ``first()`` method to your
``find()`` operation. You can use the ``sort()`` operation before selecting the first

> document to help choose the correct file.

Author: I've made some changes here. The only thing I wasn't sure what to do about is your last point:

> To perform a find operation...

If someone executes the following code:
```java
collection.find(filter)
```
Does the server execute a find operation? I understand there a distinction between a method and an operation, but the method is the vehicle by which a user causes a find operation to be performed, correct? Like are we just talking about semantics, or is this something that the reader would find confusing about how this written?

> document to help choose the correct file.

Reviewer: I appreciate you checking to make sure this is clear.

> `collection.find(filter)` ... Does the server execute a find operation? 

No, this actually happens only after a method like `first` (or: `executeExplain`, `iterator`, `cursor`, `forEach`, ...) is invoked. The preceding methods, like sort, build something like an BSON document representing the operation. Those final methods eventually send BSON to the server, where the operation is executed. Calling the `find` method on an instance of a collection will not initiate a search on the server. (Not that we need to document all of this, but the docs should not imply otherwise.)

(I do realize that some of this goes beyond what was added in this PR, so it is fine if it is addressed in another ticket.)

> document to help choose the correct file.

Author: OK, thanks for the clarification. I've made some changes.

After the owner runs this query, they find two orders that matched the
criteria.

Find Example: Full File
~~~~~~~~~~~~~~~~~~~~~~~

> The following code is a complete, standalone file that performs a find
> operation to retrieve multiple douments, and a find operation with the first
> method to retrieve a single document. 

Reviewer 2: S: I would describe this example a little bit more, maybe using a list:

```suggestion
The following code is a complete, standalone file that performs the following actions:

- Calls the ``find()`` method to retrieve multiple documents that have a ``runtime`` value less than 15, applying a projection and sort to the results 

- Calls the ``find()`` and ``first()`` method to retrieve a document that has a ``title`` value of ``"The Room"``, applying a projection and sort before returning the first match 
```

> .. include:: /includes/crud/example-intro.rst

Reviewer 2: S: this kind interrupts the flow since you have it after the code example description; maybe put it in an admonition? Or just at the beginning of this section?

Reviewer 2: Applies to all other inclusions of /includes/crud/example-intro.rst

After the owner runs this query, they find two orders that matched the
criteria.

Find Example: Full File
~~~~~~~~~~~~~~~~~~~~~~~

.. include:: /includes/crud/example-intro.rst

The following code is a complete, standalone file that performs the following
actions:

> - Calls the ``find()`` method to retrieve multiple documents that have a ``runtime`` value less than 15, applying a projection and sort to the results 
> - Calls the ``find()`` and ``first()`` method to retrieve a document that has a ``title`` value of ``"The Room"``, applying a projection and sort before returning the first match 

Reviewer 2: S:
```suggestion
- Calls the ``find()`` method to retrieve multiple documents that have a ``runtime`` value less than ``15``, applying a projection and sort to the results 
- Calls the ``find()`` and ``first()`` methods to retrieve a document that has a ``title`` value of ``"The Room"``, applying a projection and sort before returning the first match 
```

For more information about how to construct an aggregation pipeline, see
the {+mdb-server+} manual page on :manual:`Aggregation </aggregation>`.

API Documentation
-----------------

> For more information on the methods and classes used to retrieve documents
> on this page, see the following API Documentation:

Reviewer 2: S:
```suggestion
For more information about the methods and classes used to retrieve documents
on this page, see the following API documentation:
```

- `find() <{+api+}/apidocs/mongodb-driver-sync/com/mongodb/client/MongoCollection.html#find()>`__
- `first() <{+api+}/apidocs/mongodb-driver-sync/com/mongodb/client/MongoIterable.html#first()>`__
- `FindIterable <{+api+}/apidocs/mongodb-driver-sync/com/mongodb/client/FindIterable.html>`__
- `aggregate() <{+api+}/apidocs/mongodb-driver-sync/com/mongodb/client/MongoCollection.html#aggregate(java.util.List)>`__

Server Manual Entries

> ---------------------

Reviewer 2: S: outside the scope of this ticket so up to you, but I'd put this section and API Documentation as subsections under "Additional Information"

Reviewer 2: applies to all

For more information about how to construct an aggregation pipeline, see
the {+mdb-server+} manual page on :manual:`Aggregation </aggregation>`.

Additional Information
----------------------

API Documentation
~~~~~~~~~~~~~~~~~

For more information about the methods and classes used to retrieve documents

> on this page, see the following API cocumentation:

Reviewer 2: I:
```suggestion
on this page, see the following API documentation:
```
